src/frontend/hooks/useMapStyle.ts

```diff
@@ -67,45 +38,60 @@ function useLegacyStyle(): MapStyleState {
     const setStyleId = (id: string) => {
       throw new Error("Cannot set styleId on legacy map");
     };
-    if (onlineState === "unknown" || customMapState === "unknown") {
+
+    if (customMapState === "unknown") {
       return { styleType: "loading", styleUrl: null, setStyleId };
-    } else if (customMapState === "available") {
+    }
+
+    if (customMapState === "available") {
       return {
         styleType: "custom",
         styleUrl: api.getMapStyleUrl("default"),
         setStyleId,
       };
-    } else if (onlineState === "online") {
-      return { styleType: "online", styleUrl: onlineStyleURL, setStyleId };
-    } else {
-      return { styleType: "fallback", styleUrl: fallbackStyleURL, setStyleId };
     }
-  }, [onlineState, customMapState]);
+
+    return { styleType: "fallback", styleUrl: defaultMap, setStyleId };
+  }, [defaultMap, customMapState]);
 }
 
-function useMapServerStyle(): MapStyleState {
-  const [styleId, setStyleId] = React.useContext(MapStyleContext);
+function useMapServerStyle(defaultStyleUrl: string): MapStyleState {
```

Comment:
think this is missing the case where the `styleId` isn't set to a specific id (e.g. initially enabling map management).

End up with data looking like `{ styleId: "", styleUrl: "http://127.0.0.1:37916/styles/", ... }` and passing that style url to a map will cause this error to show up:

<img width="565" alt="image" src="https://user-images.githubusercontent.com/18542095/195178590-ad349ae1-7dc9-40ef-82dd-117102d38516.png">

in this case, should probably serve the default style (identical to the condition you check for on line 72). alternatively, wondering if the initial/fallback value for the `MapStyleContext` should be different? maybe not, but another potential approach

Reply:
Also having trouble re-creating this. Can you walk me through the steps you did to get here?